docs/content/docs/custom-trace.md

```diff
@@ -0,0 +1,151 @@
+  // Orchestrion may add or modify code in this function
+  // ... etc ...
+
+  //dd:ignore This particular server will NOT be instrumented
+  server := &http.Server {
+    Addr:    "127.0.0.1:8080",
+    Handler: internalServerHandler,
+  }
+
+  // Orchestrion may add or modify code further down in this function
+  // ... etc ...
+}
+```
+
+{{<callout emoji="⚠️">}}
+In certain cases, `orchestrion` adds instrumentation on the library side
+(sometimes referred to as _callee_ instrumentation; as opposed to _call site_
+instrumentation).
+
+In such cases, it is currently not possible to opt-out of instrumentation. This
+is the case for:
+- `net/http` client instrumentation
+- `github.com/gorilla/mux` middleware instrumentation
+{{</callout>}}
+
+## Creating custom trace spans
+
+Any function annotated with the `//dd:span` directive will result in a trace
+span being created around the function's execution. The directive can optionally
+provide custom span tags as `key:value` pairs (all parsed as literal strings):
+
+```go
+//dd:span tag-name:for other-tag:bar
+func tracedFunction() {
+  // This function will be represented as a span named "tracedFunction"
+}
+```
+
+### Result Capture
+
+Functions annotated with `//dd:span` which return an `error` value will
+automatically annotate spans with the returned `error` information if that is
+non-`nil`.
+
+```go
+package demo
+
+import "errors"
+
+//dd:span
+func failableFunction() (any, error) {
+  // This span will have error information attached automatically.
+  return nil, errors.ErrUnsupported
+}
+```
+
+### Operation Name
+
+The name of the operation (span name) is determined using the following
+precedence list (first non-empty is selected):
+
+- The `span.name` tag specified as a directive argument
+  ```go
+  //dd:span span.name:operationName
+  func tracedFunction() {
+    // This function will be represented as a span named "operationName"
+  }
+  ```
+- The name of the function (closures do not have a name)
+  ```go
+  //dd:span tag-name:for other-tag:bar
+  func tracedFunction() {
+    // This function will be represented as a span named "tracedFunction"
+  }
+  ```
+- The value of the very first tag from the directive arguments list
+  ```go
+  //dd:span tag-name:spanName other-tag:bar
+  tracedFunction := func() {
+    // This function will be represented as a span named "spanName"
+  }
+  ```
+
+### Trace Context Propagation
+
+If the annotated function accepts a {{<godoc "context" "Context" >}} argument,
+that context will be used for trace propagation. Otherwise, if the function
+accepts a {{<godoc "net/http" "Request" "*">}} argument, the requests's context will
+be used for trace propagation.
+
+Functions that accept neither solely rely on _goroutine local storage_ for trace
+propagation. This means that traces may be split on _goroutine_ boundaries
+unless a {{<godoc "context" "Context" >}} or {{<godoc "net/http" "Request" "*">}}
+value is passed across.
```

Comment:
Does this last sentence mean that if a context or request _is_ passed to child goroutines, that the trace _will_ be propagated? In other words, if `caller` in the example below didn't accept a context parameter, but passed one to `callee`, would the trace context be propagated?

```go
//dd:span
func caller() {
  wait := make(chan struct{}, 1)
  defer close(wait)

  ctx, cancel := context.WithTimeout(10*time.Second)
  defer cancel()
  // Is the span propagated here to callee?
  go callee(ctx, wait)
  <-wait
}
```

Reply:
Darn... I intentionally remained fuzzy on this because I hadn't tested it yet, and hoped nobody would notice 🥸

Reply:
But so in case the `ctx` is created _inside_ of the span, then no, it will not propagate context (because it does not have it).

Quite frankly, the more I look at this, the more I think @felixge was right in that we probably need to propagate the GLS context slot to new goroutines the way profiler flags are propagated... But then we suddenly need to ensure the slot's value is safe to use from multiple goroutines...